twin16: ghost sprites in miaj

rp-jt opened the pull request:
I have added the different changes commented in #929 

In the current version, boot-up tests pass more often than not, but they still appear to fail sometimes due to the objects.
Also, in Dark Adventure and Vulcan Venture sprites flicker sometimes.

In jt00778x_dma, it was originally proposed to use `oram_we <= nx_cpwa[11];` to clear the table every time, however, this brought some problems with objects in Dark Adventure and Vulcan Venture, since they actively use the dma.
By using `oram_we <= (obj_en | objbufinit) & nx_cpwa[11];`, there is always a phase of clearing the table that can be extended if `dma_on==0`. This also works for eliminating the ghost sprites.

In jttwin16_share, I created the signal `dma_busy` so oram_we is not asserted when the the dma is in use. However, I started passing `objbufinit` instead of the actual `dma_busy` because it seemed to work better in terms of flickering appearing less and more often passing the boot-up tests, but this might be wrong.

There's still some work to go

jotego commented:
I think objbufinit is shorter than even the DMA clear phase. We need to use dma_busy.
Look at when the clash happens in simulation and whether DTACKN is being held high correctly while dma_busy is set.
Also check the linter error.

rp-jt commented:
@jotego It is seen in simulation that DTACKn is held high during dma_busy when oram_cs is active

Also, fixed linter issue and changed it to use dma_busy instead of objbufinit in shared.

jotego commented:
We cannot merge it if it breaks the boot up.
